api/logs/src/main/java/io/opentelemetry/api/events/EventLogger.java

```diff
@@ -0,0 +1,144 @@
+ *   // Create EventLogger by wrapping Logger and setting event domain to "acme-observability"
+ *   private final EventLogger eventLogger =
+ *     EventLogger.create(
+ *       loggerProvider.get("instrumentation-library-name"), "acme.observability");
+ *
+ *   void doWork() {
+ *     eventLogger
+ *       .eventBuilder("my-event")
+ *       .setAllAttributes(Attributes.builder()
+ *          .put("key1", "value1")
+ *          .put("key2", "value2").build())
+ *       .emit();
+ *   }
+ * }
+ * }</pre>
+ */
+public final class EventLogger {
+
+  private final Logger logger;
+  private final String eventDomain;
+
+  private EventLogger(Logger logger, String eventDomain) {
+    this.logger = logger;
+    this.eventDomain = eventDomain;
+  }
+
+  /**
+   * Create a new {@link EventLogger}.
+   *
+   * <p>{@link #eventBuilder(String)} delegates to {@link Logger#logRecordBuilder()}. The {@code
+   * eventDomain} is included on every emitted {@code LogRecord} in the {@link
+   * SemanticAttributes#EVENT_DOMAIN} attribute.
+   *
+   * @param logger the delegate {@link Logger}
+   * @param eventDomain the event domain, which acts as a namespace for event names. Within a
+   *     particular event domain, event name defines a particular class or type of event.
+   * @return an {@link EventLogger} instance
+   */
+  public static EventLogger create(Logger logger, String eventDomain) {
+    return new EventLogger(logger, eventDomain);
+  }
+
+  /**
+   * Return an {@link EventBuilder} to emit an Event.
+   *
+   * <p>Build the event using the {@link EventBuilder} setters, and emit via {@link
+   * EventBuilder#emit()}. The {@code eventName} is included on the emitted {@code LogRecord} in the
+   * {@link SemanticAttributes#EVENT_NAME} attribute.
+   *
+   * @param eventName the event name, which acts as a classifier for events. Within a particular
+   *     event domain, event name defines a particular class or type of event.
+   */
+  public EventBuilder eventBuilder(String eventName) {
+    return new EventBuilderImpl(logger.logRecordBuilder(), eventName);
+  }
+
+  private class EventBuilderImpl implements EventBuilder {
+
+    private final LogRecordBuilder logRecordBuilder;
+    private final String eventName;
+
+    private EventBuilderImpl(LogRecordBuilder logRecordBuilder, String eventName) {
+      this.logRecordBuilder = logRecordBuilder;
+      this.eventName = eventName;
+    }
+
+    @Override
+    public LogRecordBuilder setEpoch(long timestamp, TimeUnit unit) {
+      logRecordBuilder.setEpoch(timestamp, unit);
+      return this;
+    }
+
+    @Override
+    public LogRecordBuilder setEpoch(Instant instant) {
+      logRecordBuilder.setEpoch(instant);
+      return this;
+    }
+
+    @Override
+    public LogRecordBuilder setContext(Context context) {
+      logRecordBuilder.setContext(context);
+      return this;
+    }
+
+    @Override
+    public LogRecordBuilder setSeverity(Severity severity) {
```

Comment:
Are we expecting events to have severity & severity text? Are these even valid properties of an event?

Reply:
I don't think so. Kept for sake of making this a refactor but happy to remove. Let's continue discussion in [this thread](https://github.com/open-telemetry/opentelemetry-java/pull/4961#discussion_r1024259386)!